htdocs/fourn/class/fournisseur.product.class.php

```diff
@@ -537,8 +542,8 @@ public function fetch_product_fournisseur_price($rowid, $ignore_expression = 0)
 				$this->fourn_multicurrency_id          = $obj->fk_multicurrency;
 				$this->fourn_multicurrency_code        = $obj->multicurrency_code;
 				if ($conf->barcode->enabled) {
-					$this->fourn_barcode = $obj->barcode;
-					$this->fourn_fk_barcode_type = $obj->fk_barcode_type;
+					$this->barcode = $obj->barcode;
```

Comment:
I don't think this is good.
$this is a product. So $this->barcode is the barcode of the product.
But $obj->barcode is the barcode of the supplier reference. If the product has 3 suppliers, each supplier may have a barcode different. Each barcode of supplier is stored into product_fournisseur_price so into a different table than the barcode of product we use and store into llx_product.

Reply:
Forgot about supplier product being extend from product class. 
I refactor fourn_barcode to supplier_barcode for both fetch supplier product and fetch supplier product list. Does this still apply for 13 beta?